samples/google-sheets-object-generator/sheets.js

```diff
@@ -0,0 +1,62 @@
+var Sheets = function (opts) {
+    const key = opts.key
+    const query = opts.query
+    const proxyUrl = 'https://cors-anywhere.herokuapp.com/'
+    const targetUrl = 'https://spreadsheets.google.com/tq?callback=test&key=' + key
+    let url = proxyUrl + targetUrl
+    if (query) {
+        url += '&tq=' + query
+    }
+    //alert(url)
+
+
+    this.getData = (callback) => {
+        fetch(url)
+          .then(blob => blob.text())
+          .then(response => {
+              response = convertToJSON(response)
+              callback(response)
+              //console.log(response)
+          })
+    }
+
+    const convertToJSON = (response) => {
+        // get rid of prefix: google.visualization.Query.setResponse(
+        response = response.split('google.visualization.Query.setResponse(')[1]
+        // get rid of trailing );
+        response = response.substring(0, response.length - 2)
+        response = JSON.parse(response)
+        return simplifyData(response)
+    }
+
+    const simplifyData = jsonData => {
+        const columns = jsonData.table.cols
+        const records = jsonData.table.rows
+
+        // create an array of arrays with cell data
+        const arrayData = records.map(record => {
+            return record.c.map((cell) => cell.v)
```

Comment:
Hi Ricky,

Another bug is that sometimes cell is null, but this seems to fix it:

const arrayData = records.map(record => {
    return record.c.map((cell) => cell ? cell.v : '')
})

Reply:
Ooh, good call. I’m on mobile for a while today but i’ll update ASAP